bin/manage.py

```diff
@@ -0,0 +1,406 @@
+import fcntl
+import inspect
+import json
+import logging
+import os
+import socket
+import signal
+import struct
+import sys
+import time
+
+from functools import wraps
+
+import consul as pyconsul
+#import manta
+from pymongo import MongoClient
+from pymongo.errors import *
+
+consul = pyconsul.Consul(host=os.environ.get('CONSUL', 'consul'))
+logging.basicConfig(format='%(asctime)s %(levelname)s %(name)s %(message)s',
+                    stream=sys.stdout,
+                    level=logging.getLevelName(
+                        os.environ.get('LOG_LEVEL', 'INFO')))
+
+log = logging.getLogger('manage.py')
+
+def debug(fn):
+    """
+    Function/method decorator to trace calls via debug logging.
+    Is a pass-thru if we're not at LOG_LEVEL=DEBUG. Normally this
+    would have a lot of perf impact but this application doesn't
+    have significant throughput.
+    """
+    @wraps(fn)
+    def wrapper(*args, **kwargs):
+        try:
+            # because we have concurrent processes running we want
+            # to tag each stack with an identifier for that process
+            arg = "[{}]".format(sys.argv[1])
+        except IndexError:
+            arg = "[pre_start]"
+        name = '{}{}{}'.format(arg, (len(inspect.stack()) * " "), fn.__name__)
+        log.debug('%s' % name)
+        out = apply(fn, args, kwargs)
+        log.debug('%s: %s', name, out)
+        return out
+    return wrapper
+
+def get_environ(key, default):
+    """
+    Gets an environment variable and trims away comments and whitespace.
+    """
+    val = os.environ.get(key, default)
+    try:
+        val = val.split('#')[0]
+        val = val.strip()
+    finally:
+        # just swallow AttributeErrors for non-strings
+        return val
+
+# ---------------------------------------------------------
+
+SESSION_CACHE_FILE = get_environ('SESSION_CACHE_FILE', '/tmp/mongodb-session')
+SESSION_NAME = get_environ('SESSION_NAME', 'mongodb-replica-set-lock')
+SESSION_TTL = int(get_environ('SESSION_TTL', 60))
+
+# consts for node state
+PRIMARY = 'mongodb-primary'
+SECONDARY = 'mongodb-secondary'
+REPLICA = 'mongodb'
```

Comment:
We never mark any instances as REPLICA and so this means we have an extra ContainerPilot config reload for all secondaries because they start out as showing the "mongodb" service and then need to be updated to show "mongodb-secondary". This reload makes it take a little extra time to settle.

Also, per my comments below I'm not sure we actually need to mark the primary vs secondary as separate Consul services.